lib/fs.ts

```diff
@@ -192,21 +204,29 @@ class FSPath {
 			return;
 		}
 
-		this.writeUpdateNow(dataFetcher, options);
+		return this.writeUpdateNow(dataFetcher, options);
 	}
-	finishUpdate() {
+	async finishUpdate() {
 		const pendingUpdate = pendingUpdates.get(this.path);
 		if (!pendingUpdate) throw new Error(`FS: Pending update not found`);
 		if (!pendingUpdate.isWriting) throw new Error(`FS: Conflicting update`);
 
 		pendingUpdate.isWriting = false;
 		const throttleTime = pendingUpdate.throttleTime;
 		if (!throttleTime || throttleTime < Date.now()) {
-			this.checkNextUpdate();
+			await this.checkNextUpdate();
 			return;
 		}
 
-		pendingUpdate.throttleTimer = setTimeout(() => this.checkNextUpdate(), throttleTime - Date.now());
+		let timeout: NodeJS.Timeout;
+		const promise: Promise<void> = new Promise(resolve => {
```

Comment:
I don't think this makes sense as a Promise? The timer is cancelable.

Reply:
A Promise must either resolve or reject, but this Promise isn't resolving if the timer is canceled.

I would think this should be designed in such a way that it either resolves or rejects if the timer is canceled.

Reply:
In other words, a cancelable timer is fine, but the entire cancellation process needs to be handled inside the Promise.